### PR TITLE
Add support for symfony ^7.1

### PR DESCRIPTION
**Pull request by lyrixx**
Composer outdated:
```
Direct dependencies required in composer.json:
psr/container                     1.1.2  2.0.2 Common Container Interface (PHP FIG PSR-11)
psr/http-message                  1.1    2.0   Common interface for HTTP messages
symfony/config                    6.4.8  7.1.1 Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                   6.4.10 7.1.3 Eases the creation of beautiful and testable command line interfaces
symfony/filesystem                6.4.9  7.1.2 Provides basic utilities for the filesystem
```

The two first are covered by another PR
And the last tree are blocked by https://github.com/consolidation/self-update/pull/28#issuecomment-2301634860

### CODE REVIEW

File: src/Cli/Serializer/PemNormalizer.php [hunk 1]
```diff
@@ -21,22 +21,22 @@
...
 class PemNormalizer implements NormalizerInterface, DenormalizerInterface
 {
-    public function normalize($object, ?string $format = null, array $context = [])
+    public function normalize(mixed $object, ?string $format = null, array $context = []): array|string|int|float|bool|\ArrayObject|null
     {
         return $object->getPEM();
     }
 
-    public function denormalize($data, $class, ?string $format = null, array $context = [])
+    public function denormalize(mixed $data, string $class, ?string $format = null, array $context = []): mixed
```
Comment (SamMousa) [inline src/Cli/Serializer/PemNormalizer.php:29]: Return type is actually `object`.

File: src/Cli/Serializer/PemNormalizer.php [hunk 2]
```diff
@@ -21,22 +21,22 @@
  */
 class PemNormalizer implements NormalizerInterface, DenormalizerInterface
 {
-    public function normalize($object, ?string $format = null, array $context = [])
+    public function normalize(mixed $object, ?string $format = null, array $context = []): array|string|int|float|bool|\ArrayObject|null
     {
         return $object->getPEM();
```
Comment (SamMousa) [inline src/Cli/Serializer/PemNormalizer.php:26]: Return type can be more specific.

File: src/Cli/Serializer/PemEncoder.php [hunk 3]
```diff
@@ -21,22 +21,22 @@ class PemEncoder implements EncoderInterface, DecoderInterface
...
     public const FORMAT = 'pem';
 
-    public function encode($data, $format, array $context = []): string
+    public function encode(mixed $data, string $format, array $context = []): string
     {
         return trim($data)."\n";
     }
 
-    public function decode($data, $format, array $context = [])
+    public function decode(string $data, string $format, array $context = []): mixed
```
Comment (SamMousa) [inline src/Cli/Serializer/PemEncoder.php:29]: Return type string

**Comment by SamMousa** [comment]
Some return types are not narrow enough. Note: return types can be narrowed in subclasses / implementations of interfaces.

**Review by SamMousa — CHANGES_REQUESTED** [review]

File: src/Cli/Serializer/PemNormalizer.php (hunk 1 above)
Comment (lyrixx) [reply src/Cli/Serializer/PemNormalizer.php:29]: It does not work: 

```
PHP Fatal error:  Declaration of AcmePhp\Cli\Serializer\PemNormalizer::normalize(mixed $object, ?string $format = null, array $context = []): object must be compatible with Symfony\Component\Serializer\Normalizer\NormalizerInterface::normalize(mixed $object, ?string $format = null, array $context = []): ArrayObject|array|string|int|float|bool|null in /home/gregoire/dev/github.com/acmephp/acmephp/src/Cli/Serializer/PemNormalizer.php on line 24

```

**Comment by lyrixx** [comment]
> Some return types are not narrow enough. Note: return types can be narrowed in subclasses / implementations of interfaces.

I was a bit lazy 😅 

---

I pushed a new version 👍🏼 thanks for the review

**Review by SamMousa — APPROVED** [review]

**Pull request merged**